src/dtype.rs

```diff
@@ -122,38 +122,12 @@ impl PyArrayDescr {
 ///
 /// # Custom element types
 ///
-/// You can implement this trait to manage arrays of custom element types, but they still need to be stored
-/// on Python's heap using PyO3's [Py](pyo3::Py) type.
+/// Note that it is not possible to safely store custom element types of the form `Py<T>` where `T: PyClass`
+/// as only the data type can be checked during downcasting, but not the dynamic type of each element stored in the array.
+/// This is not possible as the elements and hence their dynamic type could be changed later on via aliasing Python references to the array.
+/// Therefore, the only safe type to store in object arrays is `Py<PyAny>` also known as `PyObject`.
```

Comment:
I feel like this is a bit redundant. I note a suggestion here, but any shortening is fine for me.
```suggestion
/// Note that we cannot type-safely store `Py<T>` where `T: PyClass`, because the type information would be
/// eliminated in the resulting NumPy array.
/// In other words, objects are treated as `Py<PyAny>` (a.k.a. `PyObject`) in the array, and `Py<PyAny>` can be
/// type-safely stored.
```

Reply:
I basically committed your suggestion as I agree that the docs should be succinct. Also, the reasoning of why checking the dynamic types of the elements during extract, i.e. #257, is still available in the commit message.